src/NServiceBus.Core/Sagas/Sagas.cs

```diff
@@ -43,6 +43,11 @@ protected internal override void Setup(FeatureConfigurationContext context)
                 throw new Exception("The selected persistence doesn't have support for saga storage. Select another persistence or disable the sagas feature using endpointConfiguration.DisableFeature<Sagas>()");
             }
 
+            if (!context.Container.HasComponent<ISagaIdGenerator>())
```

Comment:
Should we use the settings instead to avoid heading further into container hell?

Persisters that wish to change the generator could just do a settings.Set(new MyCustomStrategy())? (is would that mean we have to add a feature dependency)